### PR TITLE
Closes #311 - Added ability to update price of Course

### LINKED ISSUE #311: Create CoursePriceService
Create methods to show a list of courses and update table + Unit Tests

### PR DESCRIPTION
**Pull request by OleksandrMazur**
Created PriceDTO, changed int to Integer in Course and CoursePrice, created updateCoursePrice methods in CourseService and it's implementation as well as in CourseController. And updated tests.

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/service/impl/CourseServiceImpl.java [hunk 1]
```diff
@@ -209,4 +212,18 @@ public Course addDeckToCourse(Long courseId, Long deckId) {
     public List<Course> findAllCoursesBySearch(String searchString) {
         return courseRepository.findByNameIgnoreCaseContainingOrDescriptionIgnoreCaseContaining(searchString, searchString);
     }
+
+    @Override
+    @Transactional
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/impl/CourseServiceImpl.java:233]: - lets move this logic to the updateCourse() method. Any way we will be updating the price with the name and other course fields.

- There is a bug in your code. You are using the same course id for both:

1. coursePriceRepository.findOne(courseId)

2. courseRepository.findOne(courseId)

- Please fix Sonar issues introduced you.

**Review by antonbakalets — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserve/academy/spaced/repetition/domain/CoursePrice.java [hunk 2]
```diff
@@ -13,26 +13,26 @@
     private long id;
 
     @Column(name = "price")
-    private int price;
+    private Integer price;
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/CoursePrice.java:17]: Lets keep it int, see the discussion in slack.

File: src/main/java/com/softserve/academy/spaced/repetition/controller/dto/simpleDTO/PriceDTO.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.softserve.academy.spaced.repetition.controller.dto.simpleDTO;
+
+public class PriceDTO {
+
+    private Integer price;
+
+    public PriceDTO() {
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/controller/dto/simpleDTO/PriceDTO.java:7]: I would rather introduce courseDto (with the price field of type Integer, nulls allowed) and concentrated myself on updating the CourseController.updateCourse(...) method.

File: src/main/java/com/softserve/academy/spaced/repetition/service/impl/CourseServiceImpl.java (hunk 1 above)
Comment (OleksandrMazur) [reply src/main/java/com/softserve/academy/spaced/repetition/service/impl/CourseServiceImpl.java:233]: I could move the logic to updateCourse() method, there's no implementation of updating course on web part of the project and this method is not used.

**Review by antonbakalets — APPROVED** [review]

**Review by Maks-Chornyi — APPROVED** [review]

**Review by OleksandrBartosh — APPROVED** [review]

**Review by hlibOstapenko — APPROVED** [review]

**Pull request merged**